services/terrain/src/terrain/services/permanent_id_requests.clj

```diff
@@ -38,11 +44,21 @@
              :path staging-dest})))
   data-item)
 
+(defn- validate-publish-dest
+  [{:keys [path] :as data-item}]
+  (let [publish-dest (ft/path-join (config/permanent-id-publish-dir) (ft/basename path))]
+    (when (data-info/path-exists? (config/irods-user) publish-dest)
+      (throw+ {:type :clojure-commons.exception/exists
+               :error "A folder with this name has already been published."
+               :path publish-dest})))
```

Comment:
I'm not sure this validation makes sense in `data-info`.
We have a folder in a user's home directory, and we're trying to see if a folder of the same name exists in the public `curated` folder (or in the `staging` folder) before successfully creating the request, or before attempting to create a DOI for the folder, since the folder will end up in the `curated` destination once the DOI creation process is complete.

This is all `permanent-id-request` service logic.